vterebey/budgetcalculator/budget_calculator_class.py

```diff
@@ -0,0 +1,302 @@
+        self.date = date
+
+    def _add_expense(self):
+        """
+        This function adds category and it's amount to the dictionary
+        for displaying statistics
+        """
+        if self.category not in Budget.categories:
+            Budget.categories[self.category] = self.amount
+        else:
+            Budget.categories[self.category] += self.amount
+
+    def expense_record(self):
+        """This function records expenses to a file"""
+        if int(self.amount) == self.amount:
+            self.amount = int(self.amount)
+        with open("values.csv", "a", encoding="UTF-8", newline="") as csvfile:
+            writer_ = csv.writer(csvfile)
+            writer_.writerow([self.category, self.date, self.amount])
+        Budget._add_expense(self)
+
+    def expenses_stat(self):
+        """This function dispays expenses in the form of a chart"""
+        amounts = list(Budget.categories.values())
+        # pylint: disable=W0612
+        fig1, ax1 = plt.subplots()
+        ax1.pie(
+            amounts,
+            labels=list(Budget.categories.keys()),
+            autopct="%1.1f%%",
+            shadow=False,
+            startangle=90,
+        )
+        ax1.axis("equal")
+        ax1.set_title("Statistics")
+        plt.show()
+
+    def group_by_cat(self, category):
+        """This function groups expenses by category"""
+        with open("values.csv", "r", encoding="UTF-8", newline="") as csvfile:
+            reader = csv.DictReader(csvfile, delimiter=",")
+            print(*fieldnames, sep=",")
+            for row in reader:
+                if row["Category"] == category:
+                    print(f"{row['Category']},{row['Date']},{row['Amount']}")
+        print()
+
+    def group_by_month(self, month):
+        """This function groups expenses by month"""
+        with open("values.csv", "r", encoding="UTF-8", newline="") as csvfile:
+            reader = csv.DictReader(csvfile, delimiter=",")
+            print(*fieldnames, sep=",")
+            for row in reader:
+                check_month = datetime.strptime(row["Date"], "%Y-%m-%d")
+                check_month = check_month.strftime("%Y-%m")
+                if check_month == month:
+                    print(f"{row['Category']},{row['Date']},{row['Amount']}")
+        print()
+
+    def group_by_cat_month(self, category, month):
+        """This function groups expenses by category and month"""
+        with open("values.csv", "r", encoding="UTF-8", newline="") as csvfile:
+            reader = csv.DictReader(csvfile, delimiter=",")
+            print(*fieldnames, sep=",")
+            for row in reader:
+                check_month = datetime.strptime(row["Date"], "%Y-%m-%d")
+                check_month = check_month.strftime("%Y-%m")
+                if check_month == month and row["Category"] == category:
+                    print(f"{row['Category']},{row['Date']},{row['Amount']}")
+        print()
+
+
+def welcome_menu():
```

Comment:
if it is possible to separate ui to the separate module and share it between funcs and class realization, then lets do it

Reply:
Updated